src/software/thunderscope/gl/layers/gl_draw_polygon_obstacle.py

```diff
@@ -0,0 +1,161 @@
+from software.thunderscope.binary_context_managers.full_system import ProtoUnixIO
+from software.thunderscope.gl.graphics.gl_polygon import GLPolygon
+from pyqtgraph.Qt import QtGui
+from pyqtgraph.Qt.QtCore import QTimer, Qt
+from pyqtgraph.opengl import *
+
+from proto.import_all_protos import *
+from software.py_constants import *
+from software.thunderscope.gl.helpers.observable_list import ObservableList
+from software.thunderscope.proto_unix_io import ProtoUnixIO
+
+from software.thunderscope.gl.layers.gl_layer import GLLayer
+from software.thunderscope.gl.helpers.extended_gl_view_widget import MouseInSceneEvent
+
+
+class GLDrawPolygonObstacleLayer(GLLayer):
```

Comment:
all `GLLayer` functions need to implement `refresh_graphics`: https://github.com/UBC-Thunderbots/Software/blob/2452cce548787b43469e110033f569ddc06b7416/src/software/thunderscope/gl/layers/gl_layer.py#L29